### ISSUE TITLE
Implement IconMap and DiagramMap annotations

### ISSUE THREAD

**Issue by OpenModelica-TracImporter** [OpenModelica/OpenModelica#5610]
The IconMap and DiagramMap annotations were introduced in Modelica 3.0 long time ago, but they are still not implemented in OpenModelica.

The attached package provides a simple test case. The icon of the model in Test2 should only show "M2", and the diagram should be rescaled compared to the diagram of M1.

-----------
From https://trac.openmodelica.org/OpenModelica/ticket/5610

**Comment by bilderbuchi** [comment]
FYI, this ticket has actually been reopened: https://trac.openmodelica.org/OpenModelica/ticket/5610

**Comment by bilderbuchi** [comment]
Also, this does not seem to work correctly afaict. Using a recent nightly, the model
```
model A
  Modelica.Mechanics.Translational.Interfaces.Flange_a flange_a annotation(
    Placement(visible = true, transformation(origin = {-100, -2}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-100, -2}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Mechanics.Translational.Interfaces.Flange_b flange_b annotation(
    Placement(visible = true, transformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
equation

annotation(
    Icon(graphics = {Rectangle(extent = {{-100, 100}, {100, -100}}), Text(extent = {{-60, 80}, {60, -80}}, textString = "A")}),
    uses(Modelica(version = "4.0.0")));
end A;
```
looks like this: 
![image](https://user-images.githubusercontent.com/327442/160128647-6d68dae5-4208-43e0-91cf-c0340753c66a.png)

But extending from it like this
```
model B
  extends A annotation(
    IconMap(extent = {{-50, -50}, {50, 50}}, primitivesVisible = true)
  );
equation

end B;
```
looks like this:
![image](https://user-images.githubusercontent.com/327442/160128750-2e90f1a2-7b60-4b2d-921b-e36ecb4b4aee.png)

It seems that the graphical primitives are scaled correctly (great!), but the connectors are not! The spec (18.6.3) says

> If the extent of the extends-clause defines a non-null region, the base class coordinate system is mapped to the region specified by the attribute extent, if preserveAspectRatio is true for the base class the mapping shall preserve the aspect ratio. The base class coordinate system (and preserveAspectRatio) is not inherited. 

so afaict via the coordinate system mapping, the connectors (everything that uses the coordinate system, in fact) should also be transformed.

**Comment by sbaetzing** [comment]
Setting the attribute of primitivesVisible to false is not working either at the moment.  E.g

`package TestprimitivesVisible
  model BaseModel
    annotation (Icon(coordinateSystem(preserveAspectRatio=false), graphics={Rectangle(
            extent={{-24,22},{52,-72}},
            lineColor={28,108,200},
            fillColor={0,0,0},
            fillPattern=FillPattern.Solid)}), Diagram(coordinateSystem(preserveAspectRatio=false)));
  end BaseModel;

  model ExtendedModel
    extends TestprimitivesVisible.BaseModel
                      annotation (
        IconMap(
          extent={{-100,-100},{100,100}},
          primitivesVisible=false),
        DiagramMap(
          extent={{-100,-100},{100,100}},
          primitivesVisible=true));

  end ExtendedModel;
end TestprimitivesVisible;
`

still shows the iconMap of the baseclass instead of an blank one. 
I did not open a new ticket since i guess its very similar to that one.

Best regards 

Sven

**Comment by adeas31** [comment]
The issue reported by @bilderbuchi in https://github.com/OpenModelica/OpenModelica/issues/5610#issuecomment-1079026740 is still not fixed.

**Comment by bilderbuchi** [comment]
> The issue reported by @bilderbuchi in [#5610 (comment)](https://github.com/OpenModelica/OpenModelica/issues/5610#issuecomment-1079026740) is still not fixed.

I guess that fix is included in #11688?

**Comment by casella** [comment]
I guess so 😅 

@adeas31?

**Comment by adeas31** [comment]
> > The issue reported by @bilderbuchi in [#5610 (comment)](https://github.com/OpenModelica/OpenModelica/issues/5610#issuecomment-1079026740) is still not fixed.
> 
> I guess that fix is included in #11688?

Yes.
